### DISCUSSION TITLE
How to proceed with GoFish

### DISCUSSION THREAD

**Discussion by lscheffler** [VFPX/GoFish#145] [General]
We need to discuss some points on how to proceed with GoFish.

I think we should start pointing out the big problems we have yet, not some UI problems or the like. After that we should discuss it, but for starters, just identify problems. The list below is just that what is in my mind right now.

For what I see right now
- To many versions out
- GF5 runs not in parallel with GF6
- Strange naming (GoFish5.exe)
  - How to remove GF5 stuff, if a GoFish.exe is used for GF6 (this is very odd)
- Thor delivers GoFish files (https://github.com/VFPX/Thor/issues/221)
- GF's auto hooking into Thor
- Thor installation comes with sources

**Comment by Jimrnelson** [comment]
Lutz --

Significant work has already been done on a number of these issues in your absence.

When you renamed 'GoFish5' to 'GoFish' (which made a lot of sense), you did not understand how your changes would be reflected in Thor, nor did you choose to so investigate.  Thus, two changes only relevant to Thor (the name of a Thor tool and a folder name in Thor) created a mess that is still not entirely corrected. I spent about 15-20 hours in early October trying to correct it. This was avoidable and unnecessary.

The problem now is that there are still leftovers in Thor causing confusion for users.  While 'GoFish5' no longer shows in 'Check For Updates' (as there is no longer an updater for it), it does appear in other places in Thor.

The fix that we have already implemented works like this:

When a user installs your next update to GoFish, Thor checks to see if the old folder Tools\Apps\GoFish5 exists.  If it does, it asks the user if all references to GoFish5 should be removed.  If so:
* It removes folder Tools\Apps\GoFish5.
* It removes PRGs from the Tools folder having GoFish5 in the file name.
* It changes any hot keys referring to Gofish5 to point to GoFish instead.
* It changes any reference to GoFish5 in Thor menus to GoFish.

I believe that most users have not seen this as the change was implemented after your last update to GoFish on 11/11.

***

GoFishClassic (actually GoFish 5.1.009) is available for those who had problems with the new GoFish or who chose not to upgrade to it.  The intent is that it will remain frozen without any changes.

Users have noted that they cannot use both GoFish and GoFishClassic in the same session.  

***

Any other topics you mentioned should be addressed in separate issues in the appropriate VFPX project.

**Reply by lscheffler** [reply]
This was not the question.

Anyway
1. My last contribution to this repo is 54467d68b9e, dated the 09.06.2023 (This is June). look up the history of this repo.
2. My last contribution to my fork is 69dc177bcdf8, dated 07.08.2023. Just clone my fork and lookup the history.
3. If or if I do not investigate something you are unable to know, you just see some stuff pushed, not my working branches. The whole fuzz on the Thor unzipping was around GF - and me trying trying to work it out and have a solid test base. That's why I stopped publishing changes in June. All other stuff I was unable to test, end the result was according.
4. Doug merged something 01.10.2023 in version 6.2.005 out of my fork, what I have no control of.  Probably from the push in [2]. This was a very bad situation, and there was a chance that some code that was not ready to be published was put out. I reject responsibility for any code that is moved out of my forks and inserted in others programs. 
5. I have really not followed up sine June, and all I saw was chaos I choose not to follow.
6. I have not the slightest idea what happen on the 11.11.2023.

I try to get on track again, so accusations are not very helpful. I take my part of the responsibility, but please, let's discuss this somewhere else.   
If you like an argument, open up a different discussion or we proceed by private mail, no problem.    

---
Issues? This is a discussion. The result should be to see what need to be done. And this might or might not result in issues.

Can we proceed step by step? This is now and here: Please list problems with the state of GoFish now (with or without Thor), not with persons or old commits. As simple as possible.
I'm still try to understand the recent problems, for that I ignore the part describing what is touched now. (Anyway, if it's solved, well then there is no problem? Why mind it at all? I'm happy it's solved.)

**Reply by Jimrnelson** [reply]
@lscheffler 

My sincere apologies for this.  It really never occurred to me that anybody other than you were responsible for these changes.

I am truly sorry that I offended you with this error.

/ Jim

**Comment by Jimrnelson** [comment]
@lscheffler @DougHennig 

I was addressing two of your issues, in particular the second one.  All the other bullet points should be in separate issues.

* To many versions out
* Strange naming (GoFish5.exe)
     How to remove GF5 stuff, if a GoFish.exe is used for GF6 (this is very odd)

**Reply by lscheffler** [reply]
Again, this is not an issue. This is a discussion.

![grafik](https://github.com/VFPX/GoFish/assets/27769909/3d009c5e-0103-49e4-a9fd-bfe6bd9b63f3)

Just a way to figure out where and what all the last months where about and how to proceed.

Discussions should not trigger actions. An issue might.   
Anything that we figure to be done should end up in an issue.

**Comment by DougHennig** [comment]
To address other points:

- GF5 runs not in parallel with GF6: I do not see that as a problem. Use one or the other but not both. As @Jimrnelson noted, installing GF now prompts you to remove GF 5.
- GF's auto hooking into Thor: I believe that was removed (or am I misremembering, @Jimrnelson?)
- Thor installation comes with sources: that could be cleaned up at some point but since it does not cause a problem, it is not a priority

**Reply by lscheffler** [reply]
### In parallel
- Installing in terms via Thor? GoFish is in first a stand-alone-tool that *might* integrate with Thor.
- There was some discussion on this? As expressed above, I followed all of this only with one eye.
- Removing the auto-update was an idea, for what I remember? (What basically all is I could do from my side on GF6)
### Sources  
The Thor files and the sources are very unlucky mingled. This should be done by better standard - in special since VFPXDeployment exists.

**Reply by DougHennig** [reply]
The replies are getting messy. I agree with @Jimrnelson: one discussion, one topic. Otherwise it's hard to follow what reply goes with what topic.

## In parallel:

1. Yes, when installing with Thor. If someone installs it standalone, they can manually remove GF5 if they wish, but they should not use both tools, as that causes a problem and doesn't make sense anyway.
2. @Jimrnelson and I discussed it a month or so ago via email and he implemented this.
3. If I recall correctly (too busy to look right now), the auto-registered code was removed from GF, since the updater takes care of that.

## Source

Yes, I agree, but as I said, it is not a priority since it doesn't cause any harm.

**Reply by lscheffler** [reply]
@DougHennig, @Jimrnelson  I'm puzzled too. Sure, you both are way to fast answering questions. That was not intended.   

I'm just in the *gathering questions mode*. The stuff on top are the questions / problems I saw, and I was waiting for your questions / problems / suggestions. Not your answers (yet)

**Comment by Jimrnelson** [comment]
Just back to my desk.

I'd be happy to start responding to these points if they were in separate
discussions.

On Thu, Dec 7, 2023 at 10:05 AM Doug Hennig ***@***.***>
wrote:

> To address other points:
>
>    - GF5 runs not in parallel with GF6: I do not see that as a problem.
>    Use one or the other but not both. As @Jimrnelson
>    <https://github.com/Jimrnelson> noted, installing GF now prompts you
>    to remove GF 5.
>    - GF's auto hooking into Thor: I believe that was removed (or am I
>    misremembering, @Jimrnelson <https://github.com/Jimrnelson>?)
>    - Thor installation comes with sources: that could be cleaned up at
>    some point but since it does not cause a problem, it is not a priority
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/VFPX/GoFish/discussions/145#discussioncomment-7791308>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATLLFR2CFILZ4ZRD2SFTILDYIHZNDAVCNFSM6AAAAABAK2N3SOVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TOOJRGMYDQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Nelson

**Reply by lscheffler** [reply]
@Jimrnelson IOW You have nothing to do or to suggest?

**Comment by Jimrnelson** [comment]
@lscheffler @DougHennig 

I have already responded that some of the items have already been addressed (and you had no way to know that had happened)

And the other items (from the original message in this thread) would best be discussed in separate discussion threads.  You had six different bullet points.  They are not closely related.  The discussions are best handled, in my opinion, separately.

So, for starters, that is exactly what I suggest.

**Comment by lscheffler** [comment]
I'm really sorry, but some don't grok the idea of the **discussion**. I had a certain stand of information, liked to go on recent, possibly learn what was done (what is not as simple as you like to express, thanks for the arrogance) -  or even if you liked me to participate at all, what is also within the possibilities, due to my behaviour.

It was just a try to sort it out. To learn what was done and what not.    
A simple "In general, for now this is a additional problem I see", a simple "This is done, see there", or "Make this one to an issue, we need to discuss" would have been enough. But generalizing answer are no help, I'm as wise as before.

For me this discussion makes no sense any more. One managed to spoil it unreadable to me with half answers and half accusations. 

For some it is simple to difficult to answer question by question.

The whole discussion **was intended* to have stuff related only that they belong to GoFish. To see, what is aside the [issues page](https://github.com/VFPX/GoFish/issues) important to do in this mixture of half done work, added ancient versions. To talk about a way to proceed. If this so clear to some, go ahead. I do not see it, and I do learn nothing on this.

**Comment by Jimrnelson** [comment]
@lscheffler 

Lutz --

First of all, I want to state again that I was completely wrong to think you were any way involved in the renaming of GoFish5 to GoFish.  While this did cause me a lot of unneeded work in Thor, it had nothing to do with you in any way.

I am sorry for my message impugning you in any way.  I was wrong, I ask for you to forgive this mistake.

I also hope that we can get past this.  There's a lot of work to be done.

**Reply by lscheffler** [reply]
Dear @Jimrnelson,

Must be my age - I read this the other way. I beg your pardon.

The problem is that I was preparing to do this - use GoFish instead of GoFish5. For obvious reasons. You pulled a trigger a trigger you possibly didn't notice.    
I was aware the problems and was preparing testing this, and some work was done to remove old or wrong stuff while updating GoFish via CFU. Only I was unable to proceed, there was a problem expanding stuff.

This is why I stopped in June.

Starting from this I fixed some issues, but this things where in an odd state, and I was unable to proceed. Finally I pushed the changes to my fork, let others do the merging, and I guess something on this process caused the problems. The idea was just to alter the code, but possibly the change to the appname was merged too.

From that all I noticed was havoc, and I decided not do anything on it, because the  state was to odd, and the danger to create more havoc to high.   
Because of that I just know I'm in the middle of nowhere with stuff done or not done I own unpublished code for similar solution. Half knowledge. 

So I started a discussion to figure out what I know, what is done, what need to be done. And how to proceed at all. See others POV.

This was the whole idea on this discussion. Any way, this discussion is now so crowded, there is little hope to follow it. That's why I closed it. I try to find a different way to gather the information I need. We can try to start over, if you like.

**Comment by Jimrnelson** [comment]
Yes, and there definitely was confusion about the use and understanding of the word "issues".

I have been using it (in this thread) to refer to "things to talk about".  That is, in a discussion, what are the things / topics / items/ ***issues*** to be discussed? (For me, those choices are interchangeable.)

You clearly (and reasonably) understood my reference to it as meaning "Issues on the Issues Page".

Let's start over.

***

Each of the ***items*** in you original bullet list are worthy of discussion.  I believe that there would be enough to discuss for each item to warrant separate discussions.

I do understand that you want to get a handle of where we are on each item, but I believe (knowing us) there will be a lot to say about each.  

I would find having them intertwined in one Discussion to be quite confusing.

**Reply by lscheffler** [reply]
That's a start. :)

And yes, we should not go into detail for stuff we identified worth further discussion here. This was so obvious to me, I forgot to mention. :(

What remains, is the question for other stuff that should be addressed  you or @DougHennig or anybody else know *outside normal issues* .   
And the order of things.

**Reply by Jimrnelson** [reply]
@lscheffler 

For now, we have a good starting list.  If other items  (**not calling them issues!**) for discussion come to mind, they can be added later.

Right now, I see about 25 issues on the Issue page for Thor.  Plenty to keep me busy.

I have not looked at the Issues page for GF, but would imagine there might be quite a few there as well.

**Reply by lscheffler** [reply]
GF? 34 open as of now, not all valuable.

The question that remains it, what must be done before we go to the regular issues page.
